t1/src/decoder/attribute/isFloatmul.scala

```diff
@@ -0,0 +1,345 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 Jiuyang Liu <liu@jiuyang.me>
+
+package org.chipsalliance.t1.rtl.decoder.attribute
+
+import org.chipsalliance.t1.rtl.decoder.T1DecodePattern
+
+object isFloatmul {
+  def apply(t1DecodePattern: T1DecodePattern): isFloatmul =
+    Seq(
+      y _ -> Y,
+      n _ -> N,
+      dc _ -> DC
+    ).collectFirst {
+      case (fn, tri) if fn(t1DecodePattern) => isFloatmul(tri)
+    }.get
+
+  def y(t1DecodePattern: T1DecodePattern): Boolean = {
```

Comment:
it should guarded by fp enabled, it can be accessed by `T1DecodePattern.instruction`